packages/design-system/src/atoms/buttonV2/index.tsx

```diff
@@ -118,71 +119,141 @@ const BaseButton = ({
 
   return (
     <View style={[container.alignCenter, style]}>
-      <Pressable
-        testID={testID}
-        isDisabled={isLoading || isDisabled}
-        onHoverIn={() => setIsHover(true)}
-        onHoverOut={() => setIsHover(false)}
-        onPressIn={() => setIsActive(true)}
-        onPressOut={() => setIsActive(false)}
-        onPress={onPress}
-        isHovered={isHover}
-        // Styles
-        position="relative"
-        alignItems="center"
-        justifyContent="center"
-        paddingX={`${paddingX}px`}
-        paddingY={`${paddingY}px`}
-        borderRadius={borderRadius}
-        borderColor={stylesBasedOnState.borderColor}
-        borderWidth={rest.borderWidth}
-        backgroundColor={
-          iconBackgroundColor || stylesBasedOnState.backgroundColor
-        }
-        flexDirection="row"
-        _hover={{
-          borderColor: hover.borderColor,
-          borderWidth: hover.borderWidth,
-        }}
-        _focus={{
-          backgroundColor: iconBackgroundColor || active.backgroundColor,
-        }}
-        _pressed={{
-          borderColor: active.borderColor,
-          borderWidth: active.borderWidth,
-          backgroundColor: iconBackgroundColor || active.backgroundColor,
-        }}
-      >
-        {isLoading ? (
-          <Spinner size={lineHeight} color={stylesBasedOnState.textColor} />
-        ) : (
-          <>
-            {iconVariant && variant !== 'text' && (
-              <View style={[styles.iconContainer]}>
-                <Icon
-                  variant={iconVariant}
-                  width={iconBasedOnSize[size]}
-                  height={iconBasedOnSize[size]}
-                  color={iconColor || stylesBasedOnState.textColor}
-                />
-              </View>
+      {variant === 'icon' ? (
```

Comment:
This approach includes a huge code duplication. Instead, you can do something like this:
```suggestion
  // Inside `ButtonV2` scope
  const button = (
    <Pressable
            testID={testID}
            isDisabled={isLoading || isDisabled}
            onHoverIn={() => setIsHover(true)}
            onHoverOut={() => setIsHover(false)}
            onPressIn={() => setIsActive(true)}
            onPressOut={() => setIsActive(false)}
            onPress={onPress}
            isHovered={isHover}
            // Styles
            position="relative"
            alignItems="center"
            justifyContent="center"
            paddingX={`${paddingX}px`}
            paddingY={`${paddingY}px`}
            borderRadius={borderRadius}
            borderColor={stylesBasedOnState.borderColor}
            borderWidth={rest.borderWidth}
            backgroundColor={
              iconBackgroundColor || stylesBasedOnState.backgroundColor
            }
            flexDirection="row"
            _hover={{
              borderColor: hover.borderColor,
              borderWidth: hover.borderWidth,
            }}
            _focus={{
              backgroundColor: iconBackgroundColor || active.backgroundColor,
            }}
            _pressed={{
              borderColor: active.borderColor,
              borderWidth: active.borderWidth,
              backgroundColor: iconBackgroundColor || active.backgroundColor,
            }}
          >
            {isLoading ? (
              <Spinner size={lineHeight} color={stylesBasedOnState.textColor} />
            ) : (
              <>
                {iconVariant && (
                  <View style={[styles.iconContainer]}>
                    <Icon
                      variant={iconVariant}
                      width={iconBasedOnSize[size]}
                      height={iconBasedOnSize[size]}
                      color={iconColor || stylesBasedOnState.textColor}
                    />
                  </View>
                )}
                {!iconVariant && icon && (
                  <View style={[styles.iconContainer]}>{icon}</View>
                )}
                {variant !== 'icon' && (
                  <TypographyV2
                    variant={fontBasedOnVariantAndSize}
                    selectable={false}
                    color={stylesBasedOnState.textColor}
                    style={[styles.title]}
                  >
                    {title}
                  </TypographyV2>
                )}
              </>
            )}
          </Pressable>
    )
    
    return variant === 'icon' ? <TooltipV2 title={title}>{button}</TooltipV2> : button
```